orc/src/main/java/org/apache/iceberg/orc/ORCSchemaUtil.java

```diff
@@ -326,13 +327,20 @@ private static TypeDescription buildOrcProjection(
             orcType = originalType.clone();
           }
         } else {
+          Types.NestedField field = root.findField(fieldId);
           if (isRequired) {
-            throw new IllegalArgumentException(
+            Preconditions.checkArgument(
+                field.initialDefault() != null,
+                "Missing required field: %s (%s)",
```

Comment:
Previously, the error message for a missing required field only showed the field ID. I updated it to match the other formats and to provide the full column name, which is more friendly.

This will also not fail missing required fields that have defaults. Instead, the `UnsupportedOperationException` for default support is thrown.

Reply:
I agree that the full-column name is most friendly, but I think for the other formats, we just give the field name:

https://github.com/apache/iceberg/blob/8d7ad4a7c94d1755f58d8c5f4fd4040fe4873e32/core/src/main/java/org/apache/iceberg/avro/ValueReaders.java#L263

But that's for another PR

Reply:
Yeah, I didn't realize I'd used the wrong name in the other PRs. I'll have to go fix that.